cli/tre/commands/shared_services/shared_service.py

```diff
@@ -74,16 +73,97 @@ def shared_service_invoke_action(shared_service_context: SharedServiceContext, c
+    log = logging.getLogger(__name__)
+
+    shared_service_id = shared_service_context.shared_service_id
+    if shared_service_id is None:
+        raise click.UsageError('Missing shared_service ID')
+
+    if definition is None:
+        if definition_file is None:
+            raise click.UsageError('Please specify either a definition or a definition file')
+        definition = definition_file.read()
+
+    definition_dict = json.loads(definition)
+
+    client = ApiClient.get_api_client_from_config()
+    response = client.call_api(
+        log,
+        'PATCH',
+        f'/api/shared-services/{shared_service_id}',
+        headers={'etag': etag},
+        json_data=definition_dict)
+
+    if no_wait:
+        output(response, output_format=output_format, query=query, default_table_query=r"sharedService.{id:id,name:templateName, version:templateVersion, is_enabled:isEnabled, status: deploymentStatus}")
+    else:
+        operation_url = response.headers['location']
+        operation_show(log, operation_url, no_wait=False, output_format=output_format, query=query, suppress_output=suppress_output)
+
+
+@click.command(name="set-enabled", help="Enable/disable a shared service")
+@click.option('--etag',
+              help='The etag of the shared service to update',
+              required=True)
+@click.option('--enable/--disable', is_flag=True, required=True)
+@click.option('--no-wait',
+              flag_value=True,
+              default=False)
+@output_option()
+@query_option()
+@pass_shared_service_context
+def shared_service_set_enabled(shared_service_context: SharedServiceContext, etag, enable, no_wait, output_format, query, suppress_output: bool = False):
+    log = logging.getLogger(__name__)
+
+    shared_service_id = shared_service_context.shared_service_id
+    if shared_service_id is None:
+        raise click.UsageError('Missing shared_service ID')
+
+    client = ApiClient.get_api_client_from_config()
+    click.echo(f"Setting isEnabled to {enable}...", err=True)
+    response = client.call_api(
+        log,
+        'PATCH',
+        f'/api/shared-services/{shared_service_id}',
+        headers={'etag': etag},
+        json_data={'isEnabled': enable})
+
+    if no_wait:
+        if not suppress_output or not response.is_success:
+            output(response, output_format=output_format, query=query, default_table_query=r"sharedService.{id:id,name:templateName, version:templateVersion, is_enabled:isEnabled, status: deploymentStatus}")
+    else:
+        operation_url = response.headers['location']
+        operation_show(log, operation_url, no_wait=False, output_format=output_format, query=query, suppress_output=suppress_output)
+
+
 @click.command(name="delete", help="Delete a shared_service")
 @click.option('--yes', is_flag=True, default=False)
 @click.option('--no-wait',
               flag_value=True,
               default=False)
+@click.option('--ensure-disabled',
```

Comment:
Nit: I kind of think that if I didn't know what it does and hadn't look at the code, I would have thought that an --ensure-disabled would check that the resource is disabled, and raise if it isn't. I wouldn't think that it would do the disabling
But yeah it's a nit so up to you if you act on it!